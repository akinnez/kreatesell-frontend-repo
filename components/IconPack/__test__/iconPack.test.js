import renderer from "react-test-renderer";
import { Shop, OneTimeSubscriptionIcon } from "../index";

const mockPrimaryColor = "mockColor";
const mockClassName = "mockClass";

describe("Shop :", () => {
  it("renders a snapshot of <Shop /> with mock values of primaryColor and className and the active Prop at true", () => {
    const shop = renderer
      .create(
        <Shop
          active={true}
          primaryColor={mockPrimaryColor}
          className={mockClassName}
        />
      )
      .toJSON();
    expect(shop).toMatchSnapshot();
  });
  it("renders a snapshot of <Shop /> with mock values of primaryColor and className and the active Prop at false", () => {
    const shop = renderer
      .create(
        <Shop
          active={false}
          primaryColor={mockPrimaryColor}
          className={mockClassName}
        />
      )
      .toJSON();
    expect(shop).toMatchSnapshot();
  });
});

describe("OneTimeSubscriptionIcon : ", () => {
  it("renders a snapshot of <OneTimeSubscription /> with active set to true and onHover at false", () => {
    const oneTimeSub = renderer
      .create(<OneTimeSubscriptionIcon active={true} onHover={false} />)
      .toJSON();
    expect(oneTimeSub).toMatchSnapshot();
  });
  it("renders a snapshot of <OneTimeSubscription /> with active set to false and onHover at true", () => {
    const oneTimeSub = renderer
      .create(<OneTimeSubscriptionIcon active={false} onHover={true} />)
      .toJSON();
    expect(oneTimeSub).toMatchSnapshot();
  });
  it("renders a snapshot of <OneTimeSubscription /> with both props set at false", () => {
    const oneTimeSub = renderer
      .create(<OneTimeSubscriptionIcon active={false} onHover={false} />)
      .toJSON();
    expect(oneTimeSub).toMatchSnapshot();
  });
  it("renders a snapshot of <OneTimeSubscription /> with both props set at true", () => {
    const oneTimeSub = renderer
      .create(<OneTimeSubscriptionIcon active={true} onHover={true} />)
      .toJSON();
    expect(oneTimeSub).toMatchSnapshot();
  });
});
