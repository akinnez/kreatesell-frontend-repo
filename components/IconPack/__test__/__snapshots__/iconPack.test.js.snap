// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OneTimeSubscriptionIcon :  renders a snapshot of <OneTimeSubscription /> with active set to false and onHover at true 1`] = `
<svg
  fill="none"
  height="18"
  viewBox="0 0 18 18"
  width="18"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M9.00033 0.666992C4.40033 0.666992 0.666992 4.40033 0.666992 9.00033C0.666992 13.6003 4.40033 17.3337 9.00033 17.3337C13.6003 17.3337 17.3337 13.6003 17.3337 9.00033C17.3337 4.40033 13.6003 0.666992 9.00033 0.666992ZM9.00033 15.667C5.32533 15.667 2.33366 12.6753 2.33366 9.00033C2.33366 5.32533 5.32533 2.33366 9.00033 2.33366C12.6753 2.33366 15.667 5.32533 15.667 9.00033C15.667 12.6753 12.6753 15.667 9.00033 15.667ZM9.25866 8.28366C7.78366 7.90866 7.30866 7.50033 7.30866 6.89199C7.30866 6.19199 7.96699 5.70033 9.05866 5.70033C10.2087 5.70033 10.642 6.25033 10.6753 7.06699H12.1003C12.0587 5.95033 11.3753 4.92533 10.0253 4.59199V3.16699H8.08366V4.57533C6.82533 4.84199 5.81699 5.65866 5.81699 6.91699C5.81699 8.40866 7.05866 9.15866 8.86699 9.59199C10.492 9.97533 10.817 10.5503 10.817 11.1503C10.817 11.592 10.492 12.3087 9.06699 12.3087C7.73366 12.3087 7.20866 11.7087 7.13366 10.942H5.70033C5.78366 12.3587 6.83366 13.1587 8.08366 13.417V14.8337H10.0337V13.442C11.3003 13.2003 12.3003 12.4753 12.3087 11.1337C12.3003 9.30033 10.7253 8.66699 9.25866 8.28366Z"
    fill="#0072EF"
  />
</svg>
`;

exports[`OneTimeSubscriptionIcon :  renders a snapshot of <OneTimeSubscription /> with active set to true and onHover at false 1`] = `
<svg
  fill="none"
  height="18"
  viewBox="0 0 18 18"
  width="18"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M9.00033 0.666992C4.40033 0.666992 0.666992 4.40033 0.666992 9.00033C0.666992 13.6003 4.40033 17.3337 9.00033 17.3337C13.6003 17.3337 17.3337 13.6003 17.3337 9.00033C17.3337 4.40033 13.6003 0.666992 9.00033 0.666992ZM9.00033 15.667C5.32533 15.667 2.33366 12.6753 2.33366 9.00033C2.33366 5.32533 5.32533 2.33366 9.00033 2.33366C12.6753 2.33366 15.667 5.32533 15.667 9.00033C15.667 12.6753 12.6753 15.667 9.00033 15.667ZM9.25866 8.28366C7.78366 7.90866 7.30866 7.50033 7.30866 6.89199C7.30866 6.19199 7.96699 5.70033 9.05866 5.70033C10.2087 5.70033 10.642 6.25033 10.6753 7.06699H12.1003C12.0587 5.95033 11.3753 4.92533 10.0253 4.59199V3.16699H8.08366V4.57533C6.82533 4.84199 5.81699 5.65866 5.81699 6.91699C5.81699 8.40866 7.05866 9.15866 8.86699 9.59199C10.492 9.97533 10.817 10.5503 10.817 11.1503C10.817 11.592 10.492 12.3087 9.06699 12.3087C7.73366 12.3087 7.20866 11.7087 7.13366 10.942H5.70033C5.78366 12.3587 6.83366 13.1587 8.08366 13.417V14.8337H10.0337V13.442C11.3003 13.2003 12.3003 12.4753 12.3087 11.1337C12.3003 9.30033 10.7253 8.66699 9.25866 8.28366Z"
    fill="#0072EF"
  />
</svg>
`;

exports[`OneTimeSubscriptionIcon :  renders a snapshot of <OneTimeSubscription /> with both props set at false 1`] = `
<svg
  fill="none"
  height="18"
  viewBox="0 0 18 18"
  width="18"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M9.00033 0.666992C4.40033 0.666992 0.666992 4.40033 0.666992 9.00033C0.666992 13.6003 4.40033 17.3337 9.00033 17.3337C13.6003 17.3337 17.3337 13.6003 17.3337 9.00033C17.3337 4.40033 13.6003 0.666992 9.00033 0.666992ZM9.00033 15.667C5.32533 15.667 2.33366 12.6753 2.33366 9.00033C2.33366 5.32533 5.32533 2.33366 9.00033 2.33366C12.6753 2.33366 15.667 5.32533 15.667 9.00033C15.667 12.6753 12.6753 15.667 9.00033 15.667ZM9.25866 8.28366C7.78366 7.90866 7.30866 7.50033 7.30866 6.89199C7.30866 6.19199 7.96699 5.70033 9.05866 5.70033C10.2087 5.70033 10.642 6.25033 10.6753 7.06699H12.1003C12.0587 5.95033 11.3753 4.92533 10.0253 4.59199V3.16699H8.08366V4.57533C6.82533 4.84199 5.81699 5.65866 5.81699 6.91699C5.81699 8.40866 7.05866 9.15866 8.86699 9.59199C10.492 9.97533 10.817 10.5503 10.817 11.1503C10.817 11.592 10.492 12.3087 9.06699 12.3087C7.73366 12.3087 7.20866 11.7087 7.13366 10.942H5.70033C5.78366 12.3587 6.83366 13.1587 8.08366 13.417V14.8337H10.0337V13.442C11.3003 13.2003 12.3003 12.4753 12.3087 11.1337C12.3003 9.30033 10.7253 8.66699 9.25866 8.28366Z"
    fill="#323232"
  />
</svg>
`;

exports[`OneTimeSubscriptionIcon :  renders a snapshot of <OneTimeSubscription /> with both props set at true 1`] = `
<svg
  fill="none"
  height="18"
  viewBox="0 0 18 18"
  width="18"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M9.00033 0.666992C4.40033 0.666992 0.666992 4.40033 0.666992 9.00033C0.666992 13.6003 4.40033 17.3337 9.00033 17.3337C13.6003 17.3337 17.3337 13.6003 17.3337 9.00033C17.3337 4.40033 13.6003 0.666992 9.00033 0.666992ZM9.00033 15.667C5.32533 15.667 2.33366 12.6753 2.33366 9.00033C2.33366 5.32533 5.32533 2.33366 9.00033 2.33366C12.6753 2.33366 15.667 5.32533 15.667 9.00033C15.667 12.6753 12.6753 15.667 9.00033 15.667ZM9.25866 8.28366C7.78366 7.90866 7.30866 7.50033 7.30866 6.89199C7.30866 6.19199 7.96699 5.70033 9.05866 5.70033C10.2087 5.70033 10.642 6.25033 10.6753 7.06699H12.1003C12.0587 5.95033 11.3753 4.92533 10.0253 4.59199V3.16699H8.08366V4.57533C6.82533 4.84199 5.81699 5.65866 5.81699 6.91699C5.81699 8.40866 7.05866 9.15866 8.86699 9.59199C10.492 9.97533 10.817 10.5503 10.817 11.1503C10.817 11.592 10.492 12.3087 9.06699 12.3087C7.73366 12.3087 7.20866 11.7087 7.13366 10.942H5.70033C5.78366 12.3587 6.83366 13.1587 8.08366 13.417V14.8337H10.0337V13.442C11.3003 13.2003 12.3003 12.4753 12.3087 11.1337C12.3003 9.30033 10.7253 8.66699 9.25866 8.28366Z"
    fill="#0072EF"
  />
</svg>
`;

exports[`Shop : renders a snapshot of <Shop /> with mock values of primaryColor and className and the active Prop at false 1`] = `
<div
  className="iconwrapper mockClass"
  style={
    Object {
      "backgroundColor": "",
    }
  }
>
  <svg
    fill="none"
    height="19"
    viewBox="0 0 18 19"
    width="18"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M17.8927 4.15064L15.6427 0.367012C15.5072 0.139342 15.2632 0 15 0H3C2.73677 0 2.49282 0.139342 2.35726 0.367012L0.107259 4.15064C0.0371268 4.26832 4.70335e-06 4.40304 4.70335e-06 4.54035C-0.00289371 6.20825 1.33418 7.56308 2.98725 7.56725H2.99325C3.73406 7.56916 4.44902 7.29264 4.99875 6.79161C6.13423 7.82706 7.86227 7.82706 8.99774 6.79161C10.1338 7.82934 11.8644 7.82934 13.0005 6.79161C13.88 7.59215 15.1445 7.79428 16.2262 7.30721C17.3078 6.82014 18.0032 5.73546 18 4.54035C18 4.40304 17.9629 4.26832 17.8927 4.15064Z"
      fill="#3A416F"
      opacity="0.598982"
    />
    <path
      d="M15.012 8.69084C14.3148 8.69143 13.627 8.50677 13.0027 8.15137L12.9915 8.15738C11.9271 8.76524 10.6949 8.8611 9.56849 8.42368C9.37341 8.34855 9.18297 8.25844 8.99849 8.15394L8.99024 8.15824C7.92619 8.76652 6.69393 8.86209 5.56799 8.42368C5.37316 8.34855 5.18297 8.25844 4.99874 8.15394C4.37556 8.50837 3.68903 8.69217 2.99325 8.69084C2.74408 8.68811 2.49552 8.66226 2.25 8.61352V17.2811C2.25 17.7555 2.58579 18.1401 3 18.1401H7.49999V12.986H10.5V18.1401H15C15.4142 18.1401 15.75 17.7555 15.75 17.2811V8.61009C15.5063 8.66008 15.2595 8.68708 15.012 8.69084Z"
      fill="#3A416F"
    />
  </svg>
</div>
`;

exports[`Shop : renders a snapshot of <Shop /> with mock values of primaryColor and className and the active Prop at true 1`] = `
<div
  className="iconwrapper mockClass"
  style={
    Object {
      "backgroundColor": "#0072EF",
    }
  }
>
  <svg
    fill="none"
    height="19"
    viewBox="0 0 18 19"
    width="18"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M17.8927 4.15064L15.6427 0.367012C15.5072 0.139342 15.2632 0 15 0H3C2.73677 0 2.49282 0.139342 2.35726 0.367012L0.107259 4.15064C0.0371268 4.26832 4.70335e-06 4.40304 4.70335e-06 4.54035C-0.00289371 6.20825 1.33418 7.56308 2.98725 7.56725H2.99325C3.73406 7.56916 4.44902 7.29264 4.99875 6.79161C6.13423 7.82706 7.86227 7.82706 8.99774 6.79161C10.1338 7.82934 11.8644 7.82934 13.0005 6.79161C13.88 7.59215 15.1445 7.79428 16.2262 7.30721C17.3078 6.82014 18.0032 5.73546 18 4.54035C18 4.40304 17.9629 4.26832 17.8927 4.15064Z"
      fill="#ffffff"
      opacity="0.598982"
    />
    <path
      d="M15.012 8.69084C14.3148 8.69143 13.627 8.50677 13.0027 8.15137L12.9915 8.15738C11.9271 8.76524 10.6949 8.8611 9.56849 8.42368C9.37341 8.34855 9.18297 8.25844 8.99849 8.15394L8.99024 8.15824C7.92619 8.76652 6.69393 8.86209 5.56799 8.42368C5.37316 8.34855 5.18297 8.25844 4.99874 8.15394C4.37556 8.50837 3.68903 8.69217 2.99325 8.69084C2.74408 8.68811 2.49552 8.66226 2.25 8.61352V17.2811C2.25 17.7555 2.58579 18.1401 3 18.1401H7.49999V12.986H10.5V18.1401H15C15.4142 18.1401 15.75 17.7555 15.75 17.2811V8.61009C15.5063 8.66008 15.2595 8.68708 15.012 8.69084Z"
      fill="#ffffff"
    />
  </svg>
</div>
`;
